### ISSUE TITLE
Note-to-Self: Contact is not a Signal user

### ISSUE THREAD

**Issue by Aridor2006** [signalapp/Signal-iOS#4740]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
When sending a note to myself, it errors out with "Send Failed" and if I tap the message it says "Contact is not a Signal user". Sending and receiving to other people works. I am also able to verify myself.

### Steps to reproduce
- Open Note To Self
- Send a message

**Actual result:** Message get's marked as Send Failed

**Expected result:** Message should be delivered without error

### Screenshots
![viber_image_2021-01-14_11-08-56](https://user-images.githubusercontent.com/3269398/104576835-fefc7200-5658-11eb-8b11-f06fc72c4ce6.jpg)

### Device info
**Device**: iPhone 11 Pro

**iOS version**: 14.2

**Signal version:** 5.1.0.31

### Link to debug log

https://debuglogs.org/fcf6c2f3d9a314169aa3f769b4940d7c0b4fce2e4f6c06abcc560d45b7cf08af.zip

**Comment by celinajean** [comment]
Same case on below link:
https://github.com/signalapp/Signal-iOS/issues/4731

**Comment by AppiePau** [comment]
Here are my two cents... I can attest the same problems described (Self note and sharing messages across devices. I noticed that my linked devices WILL receive message I type to myself and contacts using my phone, but not the other way. So all my linked deviced show the sent message from my phone, but when I send them from my PC.

However, my linked iPad seems to work fine both ways. It only occurs on my linked PC and Macbook. 

Further more, this worked until I linked a second phone and tried to switch back to my old one. That mistake was on my side. After that I completely cleaned out my devices (uninstalling Signal, cleaning data folders) and reinstalling. Even deleting and reinitiating my account did not help.

I read somewhere that this might be timing issue due to different timeservers used on different devices. I have tried tried to let my PC with different NTP servers and then relinking the devices. No cigar.

Hope this helps.
